### PR TITLE
relocates lists and associated relationship retrieval from M365

### PR DESCRIPTION
**Pull request by HiteshRepo**
removes retrieval lists and its relationships from collections/sites package
add retrieval lists and its relationships to /api/lists client wrapper

apart from above, this PRs accumulates changes from other chained PRs:
for restore backups
- https://github.com/alcionai/corso/pull/4852
- https://github.com/alcionai/corso/pull/4853
- https://github.com/alcionai/corso/pull/4854
- https://github.com/alcionai/corso/pull/4855
- https://github.com/alcionai/corso/pull/4870

for list backups
- https://github.com/alcionai/corso/pull/4786
- https://github.com/alcionai/corso/pull/4787
- https://github.com/alcionai/corso/pull/4815
- https://github.com/alcionai/corso/pull/4847
- https://github.com/alcionai/corso/pull/4848
- https://github.com/alcionai/corso/pull/4849
- https://github.com/alcionai/corso/pull/4850
- https://github.com/alcionai/corso/pull/4851
- https://github.com/alcionai/corso/pull/4871
- https://github.com/alcionai/corso/pull/4893

#### Does this PR need a docs update or release note?
- [x] :clock1: Yes, but in a later PR

#### Type of change

- [x] :sunflower: Feature

#### Issue(s)
#4754

#### Test Plan

- [x] :muscle: Manual
- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/pkg/services/m365/api/lists.go [hunk 1]
```diff
@@ -62,3 +63,255 @@ func (c Lists) PostDrive(
...
+	}
+
+	return cols, cTypes, lItems, nil
+}
+
+// getListItems utility for retrieving ListItem data and the associated relationship
+// data. Additional call append data to the tracked items, and do not create additional collections.
+// Additional Call:
+// * Fields
+func (c Lists) getListItems(ctx context.Context, siteID, listID string) ([]models.ListItemable, error) {
```
Comment (ryanfkeepers) [inline src/pkg/services/m365/api/lists.go:134]: Can you please asdd TODO comments for all `get*Items` funcs that they need to be transferred to the current pager implementation?

File: src/pkg/services/m365/api/lists_test.go [hunk 2]
```diff
@@ -55,3 +58,184 @@ func (suite *ListsAPIIntgSuite) TestLists_PostDrive() {
...
+					"itemName": "item1",
+					"itemSize": 4,
+				}
+				fields.SetAdditionalData(fieldsData)
+
+				interceptV1Path(
+					"sites",
+					siteID,
+					"lists",
+					listID).
```
Comment (ryanfkeepers) [inline src/pkg/services/m365/api/lists_test.go:128]: for our code standards, you can inline key:value pairs for readability.  Ex:
```suggestion
					"sites", siteID,
					"lists", listID).
```

File: src/pkg/services/m365/api/lists_test.go [hunk 3]
```diff
@@ -55,3 +58,184 @@ func (suite *ListsAPIIntgSuite) TestLists_PostDrive() {
...
+			defer gock.Off()
+			test.setupf()
+
+			list, err := suite.its.gockAC.Lists().GetListByID(ctx, siteID, listID)
+			test.expect(t, err)
+			assert.Equal(t, listID, *list.GetId())
+
+			items := list.GetItems()
+			assert.NotEmpty(t, items)
+			assert.Equal(t, 1, len(items))
```
Comment (ryanfkeepers) [inline src/pkg/services/m365/api/lists_test.go:212]: This is a redundant pair of checks.  Only the `Equal` is useful.

**Review by ryanfkeepers — APPROVED** [review]

**Review by ryanfkeepers — CHANGES_REQUESTED** [review]
@HiteshRepo there's been substantial code additions, including new logical processes, introduced to this PR since my last review.  Instead of adding those changes into this PR, please move them into a separate PR that I can review independently.

**Comment by HiteshRepo** [comment]
> @HiteshRepo there's been substantial code additions, including new logical processes, introduced to this PR since my last review. Instead of adding those changes into this PR, please move them into a separate PR that I can review independently.

@ryanfkeepers the additions to this PR has only been approved PR merges which were already reviewed and approved. I am using this as a main branch for 'Sharepoint Lists'. The idea is to move backup and restore functionality altogether to the main branch.

**Review by meain — APPROVED** [review]
Approving as the individual PRs were separately looked at and approved.

**Comment by HiteshRepo** [comment]
> Approving as the individual PRs were separately looked at and approved.

yes.all chained PRs have been approved and merged into this. Added them to the description.

**Comment by HiteshRepo** [comment]
> > Approving as the individual PRs were separately looked at and approved.
> 
> yes.all chained PRs have been approved and merged into this. Added them to the description.

@ryanfkeepers can we merge this? The other PRs got chained from this and this has grown big as a result.

**Comment by meain** [comment]
> ryanfkeepers can we merge this? The other PRs got chained from this and this has grown big as a result.

Let's go ahead and merge this as this PR only contains code from other PRs that have been already reviewed. Next time, let's make sure to merge them one-by-one instead of combining into a single PR and merging to main. This also help making bisect more useful.

**Comment by HiteshRepo** [comment]
> > ryanfkeepers can we merge this? The other PRs got chained from this and this has grown big as a result.
> 
> Let's go ahead and merge this as this PR only contains code from other PRs that have been already reviewed. Next time, let's make sure to merge them one-by-one instead of combining into a single PR and merging to main. This also help making bisect more useful.

In case of regression(if any), this PR will cause difficult to hunt the cause. This has been broken down into multiple PRs of logical grouped changes.
- https://github.com/alcionai/corso/pull/4916
- https://github.com/alcionai/corso/pull/4917
- https://github.com/alcionai/corso/pull/4919
- https://github.com/alcionai/corso/pull/4920
- https://github.com/alcionai/corso/pull/4922
- https://github.com/alcionai/corso/pull/4923
- https://github.com/alcionai/corso/pull/4924

will close this once all of them are merged

**Comment by HiteshRepo** [comment]
> > > ryanfkeepers can we merge this? The other PRs got chained from this and this has grown big as a result.
> > 
> > 
> > Let's go ahead and merge this as this PR only contains code from other PRs that have been already reviewed. Next time, let's make sure to merge them one-by-one instead of combining into a single PR and merging to main. This also help making bisect more useful.
> 
> In case of regression(if any), this PR will cause difficult to hunt the cause. This has been broken down into multiple PRs of logically grouped approved changes.
> 
> * [adds ListInfo details to SharepointInfo details #4916](https://github.com/alcionai/corso/pull/4916)
> * [adds list client apis and pagers #4917](https://github.com/alcionai/corso/pull/4917)
> * [adds list backup and restore handlers #4919](https://github.com/alcionai/corso/pull/4919)
> * [enables sharepoint to use lists backup handler for lists ops #4920](https://github.com/alcionai/corso/pull/4920)
> * [enable restore path creation for sharepoint lists #4922](https://github.com/alcionai/corso/pull/4922)
> * [enables sharepoint to use lists restore handler for lists ops #4923](https://github.com/alcionai/corso/pull/4923)
> * [cli changes for sharepoint lists #4924](https://github.com/alcionai/corso/pull/4924)
> 
> will close this once all of them are merged

closing this as all PRs have been merged

**Pull request closed without merging**